activerecord/lib/active_record/associations/join_dependency/join_association.rb

```diff
@@ -54,12 +54,18 @@ def join_constraints(foreign_table, foreign_klass, node, join_type, tables, scop
             end
             scope_chain_index += 1
 
-            relation = ActiveRecord::Relation.create(
-              klass,
-              table,
-              predicate_builder,
-            )
-            scope_chain_items.concat [klass.send(:build_default_scope, relation)].compact
+            klass_scope =
+              if klass.current_scope
+                klass.current_scope.clone
+              else
+                relation = ActiveRecord::Relation.create(
+                  klass,
+                  table,
+                  predicate_builder,
+                )
+                klass.send(:build_default_scope, relation)
+              end
```

Comment:
I tried some code using it, but all of them broke the tests.
Though I don't still look into it well, I have no idea to make use of `scope_attributes?`. `scope_attributes?` doesn't distinguish the case to return current_scope and one to return the scope in L61-66, does it?